src/board.rs

```diff
@@ -87,40 +87,104 @@ impl Board {
         Ok(())
     }
 
-    fn add_moves_in_direction(&mut self, start: Position, m: Move) {
-        let mut position = start + m;
-        while Self::is_in_bounds(position).is_ok() {
-            if let Some(piece) = self.get_piece(position) {
-                // allow capturing an opponent's piece
-                if piece.get_player() != self.player {
-                    self.moves.insert((start, position));
+    fn add_pawn_advance_moves(&mut self, start: Position, player: Player) {
+        let direction = match player {
+            Player::White => -1,
+            Player::Black => 1,
+        };
+        let advance_moves = match player {
+            Player::White => Move::get_pawn_advance_moves_white(),
+            Player::Black => Move::get_pawn_advance_moves_black(),
+        };
+
+        for &m in advance_moves {
```

Comment:
Since we know there is only 1 advance move direction, do we need a for loop here?

Reply:
Maybe it's better this way since it's a slice idk

Reply:
No you're right, we can return a Move instead of a slice

Reply:
We'll return `&[Move]` for convenience for now